src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -46,18 +50,90 @@ protected virtual CompletionItemRules GetCompletionItemRules(IReadOnlyList<ISymb
         private ImmutableArray<CompletionItem> CreateItems(
             ImmutableArray<ISymbol> symbols,
             SyntaxContext context,
-            bool preselect)
+            bool preselect,
+            ImmutableArray<ITypeSymbol> inferredTypes)
+        {
+            if (IsTargetTypeCompletionFilterExperimentEnabled(context.Workspace))
+            {
+                var symbolGroups = from symbol in symbols
+                                   let texts = GetDisplayAndSuffixAndInsertionText(symbol, context)
+                                   group symbol by texts into g
+                                   select g;
```

Comment:
>; [](start = 43, length = 1)

Do you want ToArray() here?

Reply:
I don't think so... I'm just going to enumerate over the results and add them each to the immutablearraybuilder. No need to index into it or realize it in advance.

Reply:
I'd prefer to have ToArray here. In case of debugging, it would be easier.

---
In reply to: [275562908](https://github.com/dotnet/roslyn/pull/34655#discussion_r275562908) [](ancestors = 275562908)